script.js

```diff
@@ -1,12 +1,365 @@
+    return uniqueGenre.indexOf(genre) == position;
+  });
+  return uniqueArray;
+}
+function addGenresToSelectGenreElement() {
+  uniqueArray = findGenres();
+  uniqueArray.forEach((genre) => {
+    let option = document.createElement("option");
+    option.setAttribute("value", genre);
+    option.innerText = genre;
+    selectGenre.appendChild(option);
+  });
+  return uniqueArray;
+}
+//Creating a function to enable element
+function enableElement(element) {
+  element.disabled = false;
+}
+//Creating a function to sort shows by name
+function sortShowByName(allShows) {
+  allShows.sort((firstname, nextname) =>
+    firstname.name.localeCompare(nextname.name)
+  );
+}
+//function to create options for select element
+function addShowsToSelectShowElement(allShows) {
+  sortShowByName(allShows);
+  allShows.forEach((show) => {
+    let option = document.createElement("option");
+    option.setAttribute("value", show.name);
+    option.innerText = show.name;
+    selectShow.appendChild(option);
+  });
+}
+let dataEpisode;
+//Function to fetch data for seasons by using show id
+function fetchAndDisplaySeason(showId) {
+  fetch("https://api.tvmaze.com/shows/" + showId + "/episodes")
+    .then((response) => response.json())
+    .then((data) => {
+      //displaying divisions by using data fetched
+      displayShowsEpisodes(data);
+      //adding episodes names season info to select element
+      addEpisodesToSelectEpisodeElement(data);
+      //dataEpisode will be used in search
+      dataEpisode = data;
+      isDisplayAllShows = false;
+    })
+    .catch((error) => console.log(error));
+}
+
+let castInfo = document.createElement("div");
+castInfo.classList.add("cast-info");
+let castDiv = document.createElement("div");
+episodeContainer.appendChild(castDiv);
+castDiv.classList.add("cast-div");
+castDiv.innerHTML = `<p>See Cast List</p>`;
+//Function to fetch cast list by showId
+function fetchCastData(showId) {
+  let url = "http://api.tvmaze.com/shows/" + showId + "/cast";
+  fetch(url)
+    .then((response) => response.json())
+    .then((data) => {
+      castInfo.innerHTML = ``;
+      castDiv.appendChild(castInfo);
+      // castDiv.classList.add("cast");
+      data.forEach((show) => {
+        let cast = document.createElement("div");
+        cast.classList.add("cast");
+        castInfo.appendChild(cast);
+
+        let image = document.createElement("img");
+        let name = document.createElement("p");
+        cast.appendChild(image);
+        cast.appendChild(name);
+        if (image) {
+          image.src = show.person.image.medium;
+        }
+        name.innerText = show.person.name;
+      });
+    });
+}
+//Adding event listener to select show element
+selectShow.addEventListener("change", function eventHandler(e) {
+  isDisplayAllShows = false;
+  castDiv.innerHTML = `<p>See Cast List</p>`;
+  episodeSection.innerHTML = ``;
+  if (e.target.value === "Please choose a show") {
+    displayShowsEpisodes(allShows);
```

Comment:
`if (e.target.value === "Please choose a show")` Is there any possibility to check the condition of ` selectedIndex = 0` for the selectShow input instead of checking the text "Please choose a show"?